src/runtime/virtcontainers/clh.go

```diff
@@ -395,8 +425,33 @@ func (clh *cloudHypervisor) CreateVM(ctx context.Context, id string, network Net
 
 	// Create the VM memory config via the constructor to ensure default values are properly assigned
 	clh.vmconfig.Memory = chclient.NewMemoryConfig(int64((utils.MemUnit(clh.config.MemorySize) * utils.MiB).ToBytes()))
-	// shared memory should be enabled if using vhost-user(kata uses virtiofsd)
-	clh.vmconfig.Memory.Shared = func(b bool) *bool { return &b }(true)
+
+	if !clh.config.BootFromTemplate {
+		// shared memory should be enabled if using vhost-user(kata uses virtiofsd)
+		// or VM templating builds the first VM with file-backed memory
+		clh.vmconfig.Memory.Shared = func(b bool) *bool { return &b }(true)
+	}
+
+	// VM templating builds VM with file-backed memory
+	if clh.config.BootToBeTemplate || clh.config.BootFromTemplate {
+		clh.vmconfig.Memory.Size = 0 
+		// virtio-mem is the only way of resizing a memory zone
+		var hotplugMethod string = "VirtioMem"
+		clh.vmconfig.Memory.HotplugMethod = &hotplugMethod
+		memoryZoneConfig := chclient.NewMemoryZoneConfig("mem0", int64((utils.MemUnit(clh.config.MemorySize) * utils.MiB).ToBytes()))
```

Comment:
Please can you create a `const` for "mem0" since it's used multiple times and if we needed to change that name in the future, having a single definition will help us avoid bugs.